engine/src/main/java/org/terasology/rendering/world/WorldRendererImpl.java

```diff
@@ -374,19 +351,7 @@ public void render(RenderingStage renderingStage) {
     private void renderShadowMap() {
         if (renderingConfig.isDynamicShadows() && isFirstRenderingStageForCurrentFrame) {
```

Comment:
Ok, this is a tricky one. Ultimately, we want the isDynamicShadows() check to NOT occur. If dynamicShadows is activated, the ShadowMapNode is added in the graph and gets processed. If it isn't, there is no ShadowMapNode in the graph. Similarly, the whole renderShadowMap() method should cease to exist. Initially it could be replaced by the shadowMapNode.render() method, eventually even that will be called through something like "RenderPipeline.transverse() or .process() or .execute() or .render()".

What I don't know is if it's worth taking care of it now already. Perhaps not yet. Perhaps it can wait until the next PR. What do you think?

What might be worth is to move the second check, isFirstRenderingStageForCurrentFrame, within the shadowMapNode, until we deal with VR support.